Update bonds when atoms moved individually

cajfisher opened the pull request:
Refreshes bonds in model window when atom coordinates changed in Model:Editing box (same as already done for groups of selected atoms).
Also updates dipole in Model:Content box as soon as changes made.

cajfisher committed:
Update bonds when atoms moved individually

Refreshes bonds in model window when atom coordinates changed in Model:Editing box (same as already done for groups of selected atoms).

cajfisher committed:
Add reader and writer for XCrysDen format (.xsf)

Enables reading and writing of structures in xsf format (including animation files)

cajfisher committed:
Fix bug in meta file reader

Fixes bug in file_meta.c when parsing units
Replaces deprecated g_pattern_match_string with g_pattern_spec_match_string in file.c
Replace QE with Quantum Espresso in file type lists for clarity
Correct various spelling/punctuation errors

cajfisher committed:
Remove moldy files specs

ovhpa commented:
Hello,
I'm back from the hectic times...
I now work on other projects, but I will come here on GDIS sometimes ;)

About the PR:

- [x] gcc and other compilers (icc, clang)
- [x] valgrind / libasan memory checks
- [x] static analysis
- [ ] review

I will review the XCrysDen file reader later with the other PR.
As per @arohl [suggestion](https://github.com/arohl/gdis/pull/4#issuecomment-368304035):
Consider changing the header of the new file you created (file_xsf.c) to be under your name, for example:
```
/*
Copyright (C) 20XX by MYNAME

myname@myprovider<mailto:mymail@myprovider>

This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 2
of the License, or (at your option) any later version.
```
Where MYNAME is.. your name :laughing: 

Best,

cajfisher committed:
Update copyright statement

cajfisher committed:
Fix bug in cif reader

ovhpa commented on src/file_cif.c:
```diff
@@ -182,7 +182,7 @@ return(0);
 /* CIF loading */
 /***************/
-#define DEBUG_LOAD_CIF 0
+#define DEBUG_LOAD_CIF 1
```
DEBUG should be OFF by default, I think.

ovhpa commented on src/spatial.c:
```diff
@@ -399,8 +399,8 @@ struct core_pak *core1, *core2, *core3;
 /* shouldn't matter when this routine is properly implemented, as we'll go */
-/* through the atom_list in the usual fashoin & terminate when it is NULL, */
-/* not when it's data ptr is NULL */ 
+/* through the atom_list in the usual fashoin and terminate when it's NULL, */
```
s/fashoin/fashion/

cajfisher committed:
Add surface dipole option

Adds a button to "Region" window of "Model editing" dialog to calculate surface dipole of a 2D model and report it in message box

cajfisher committed:
Check GLIB version

Some OSs use older versions of glib, before g_pattern_spec_match_string replaced g_pattern_match_string (deprecated since v2.70)
